resources/js/components/SiteSharingCard.vue

```diff
@@ -0,0 +1,290 @@
+        }),
+        showShareForm: false,
+      }
+    },
+
+    mounted() {
+      this.loadSharedUsers()
+      this.loadInvitations()
+    },
+
+    methods: {
+      async loadSharedUsers() {
+        try {
+          const {data} = await axios.get(`/web/user-sites/shared`, {
+            params     : { site_id: this.site.id },
+            cancelToken: new axios.CancelToken(c => this.request = c),
+          })
+
+          this.sharedWith = data
+          this.loading  = false
+        } catch (e) {
+          if (!axios.isCancel(e)) {
+            if (e.response && e.response.status === 403) {
+              this.$alert('You are not authorized to complete this action.')
+            } else if (e.response && e.response.status === 422) {
+              const errors = new Errors(e.response.data)
+              this.$alert(errors.toArray().join(' '))
+            } else {
+              this.$alert('Unable to process your request at this time. Please try refreshing the page or contact us.')
+            }
+            console.error(e)
+          }
+
+          this.loading = false
+        }
+      },
+
+      async loadInvitations() {
+        try {
+          const {data} = await axios.get(`/web/site-invitations`, {
+            params: {
+              site_id: this.site.id,
+            }
+          })
+          this.invitations = data
+        } catch (e) {
+          if (e.response && e.response.status === 403) {
+            this.$alert('You are not authorized to complete this action.')
+          } else if (e.response && e.response.status === 422) {
+            const errors = new Errors(e.response.data)
+            this.$alert(errors.toArray().join(' '))
+          } else {
+            this.$alert('Unable to process your request at this time. Please try refreshing the page or contact us.')
+          }
+          console.error(e)
+        }
+
+        this.loading = false
+      },
+
+      async inviteUser() {
+        try {
+          await this.userForm.post(`/web/sites/${this.site.id}/invitations`)
+          this.$notify({
+            type: 'success',
+            text: 'Invitation sent successfully.',
+          })
+          this.user = null
+          this.loadInvitations()
+        } catch (e) {
+          if (e.response && e.response.status === 403) {
+            this.$alert('You are not authorized to complete this action.')
+          } else if (e.response && e.response.status === 422) {
+            const errors = new Errors(e.response.data)
+            this.$alert(errors.toArray().join(' '))
+          } else {
+            this.$alert('Unable to process your request at this time. Please try refreshing the page or contact us.')
+          }
+          console.error(e)
+        }
+      },
+
+      async toggleEdit(user) {
+        try {
+          const {data} = await axios.put(`/web/user-sites/site/${this.site.id}/user/${user.id}/toggle-edit`)
+          user.can_edit = data.editable
```

Comment:
What are we trying to do here? if you want to update the user instance without having to reload, then you have to alter the array itself. This is because Vue only watches objects in a shallow manner. ie, it can't detect changes to a property of an object that is part of an array. Therefore, the UI won't always get updated using this method.

```suggestion
          this.sharedWith = this.sharedWith.map(u => u.id === user.id ? {...u, can_edit: data.editable} : u)
```